compiler/src/codegen/compcore.re

```diff
@@ -360,384 +362,467 @@ let init_codegen_env = () => {
 /* Seems a little silly when named this way, but it makes a little more sense in the
    context of this file, since we are encoding the given OCaml string in a WASM-compatible format
    (its UTF-8 byte sequence). */
-let encode_string: string => list(int) = (Utf8.decode: string => list(int));
+let encode_string: string => list(int) = Utf8.decode;
 
 let encoded_int32 = n => n * 2;
 
-let const_int32 = n =>
-  add_dummy_loc(Values.I32Value.to_value @@ Int32.of_int(n));
-let const_int64 = n =>
-  add_dummy_loc(Values.I64Value.to_value @@ Int64.of_int(n));
-let const_float32 = n =>
-  add_dummy_loc(Values.F32Value.to_value @@ Wasm.F32.of_float(n));
-let const_float64 = n =>
-  add_dummy_loc(Values.F64Value.to_value @@ Wasm.F64.of_float(n));
+let const_int32 = n => Literal.int32(Int32.of_int(n));
+let const_int64 = n => Literal.int64(Int64.of_int(n));
+let const_float32 = n => Literal.float32(n);
+let const_float64 = n => Literal.float64(n);
 
 /* These are like the above 'const' functions, but take inputs
    of the underlying types instead */
-let wrap_int32 = n => add_dummy_loc(Values.I32Value.to_value(n));
-let wrap_int64 = n => add_dummy_loc(Values.I64Value.to_value(n));
-let wrap_float32 = n => add_dummy_loc(Values.F32Value.to_value(n));
-let wrap_float64 = n => add_dummy_loc(Values.F64Value.to_value(n));
+let wrap_int32 = n => Literal.int32(n);
+let wrap_int64 = n => Literal.int64(n);
+let wrap_float32 = n => Literal.float32(n);
+let wrap_float64 = n => Literal.float64(n);
 
-let grain_number_max = 0x3fffffff;
-let grain_number_min = (-0x3fffffff); /* 0xC0000001 */
+let grain_number_max = 1073741823;
+let grain_number_min = (-1073741823); /* 0xC0000001 */
```

Comment:
Note to self to change these back (I guess refmt didn't like them)

Reply:
+1